src/utils/dataStructureUtils.ts

```diff
@@ -0,0 +1,31 @@
+function createMapFromObject(obj: Record<string, any>) {
+  const map = new Map<string, any>();
+
+  for (const key in obj) {
+    map.set(key, obj[key]);
+  }
+
+  return map;
+}
+
+function isArrayOfString(x: unknown): boolean {
+  return (
+    Array.isArray(x) &&
+    x.every(item => Boolean(item && typeof item === "string"))
+  );
+}
+
+function isArrayOfObject(x: unknown): boolean {
+  return (
+    Array.isArray(x) &&
+    x.every(item => Boolean(item && typeof item === "object"))
+  );
+}
+
+function isObjectEmpty(obj: unknown): boolean {
```

Comment:
I'm a little bit lost. Didn't you already defined this function on `lib/utils/dataStructureUtils.js`. Is `/lib/` output target? If yes, shouldn't we add lib to `.gitignore`?

Reply:
yes `lib` is the output. I'm not really sure how would npm packages work if we ignored this files?

cc: @edizcelik @bilgeyucel

Reply:
So I've investigated this. I will open another pr about this.

Way we should be doing is

1. Add `/lib` to gitignore
1. Add `"files": ["lib/**/*"],` to `package.json`

Now npm respects the gitginore file but whitelists the lib folder.